core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -3831,6 +3831,15 @@ public static List distinct(List list) {
     return Collections.nCopies(numberOfElement, element);
   }
 
+  /** Support the MAP_ENTRIES function. */
+  public static List mapEntries(Map<Object, Object> map) {
+    List result = new ArrayList(map.size());
```

Comment:
In Calcite, variables should be defined as final as much as possible.

Can `final List result = new ArrayList(map.size())` work?

Reply:
fixed